examples/randomized-benchmark/two-qubit-rb/example_2qb_rb.py

```diff
@@ -0,0 +1,232 @@
+        super().__init__(aliases)
+
+    def q(self, qubit, channel=None):
+        if qubit == "q0":
+            """
+            Put here all commands related to the targeted channel (eg. using two different elements for addressing the 
+            qubit with two distinct frequencies, this can be specified through the channel argument, or any additional 
+            that might be required)
+            """
+            return qubit
+
+        elif qubit == "q1":
+            return qubit
+        else:
+            return qubit
+
+    def res(self, resonator):
+        if resonator == "rr1":
+            return resonator
+        elif resonator == "rr2":
+            return resonator
+
+    def coupler(self, qubit1, qubit2):
+        if qubit1 == 'q0' and qubit1 == 'q1':
+            return "coupler" # Return name of coupling element for your set of qubits
+        else:
+            return coupler
+
+    @property
+    def aliases(self):
+        return self._aliases
+
+
+# Baking Macros required for two qubit gates
+
+# Here is my assumed native two qubit gate
+
+
+resolve = RBTwoQubitResolver()
+
+
+def CZ(b_seq: Baking, ctrl: str = "q0", tgt: str = "q1"):
+    """
+    The native two-qubit gate shall always contain align before and after the playing statements
+    """
+    b_seq.align(resolve.q(ctrl), resolve.q(tgt), resolve.coupler(ctrl, tgt))
+
+    b_seq.play("CZ", resolve.coupler(ctrl, tgt))
+
+    b_seq.align(resolve.q(ctrl), resolve.q(tgt), resolve.coupler(ctrl, tgt))
+
+
+def CNOT(b_seq: Baking, ctrl: str = "q0", tgt: str = "q1"):
+    # Map your pulse sequence for performing a CNOT using baking play statements
+    #
+
+    # Option 1: simple play statement for single qubit gate
+
+    mY_2(b_seq, tgt)
+    CZ(b_seq, ctrl, tgt)
+    Y_2(b_seq, tgt)
+
+
+def iSWAP(b_seq: Baking, ctrl: str = "q0", tgt: str = "q1"):
+    mX_2(b_seq, tgt)
+    CNOT(b_seq, tgt, ctrl)
+    mX_2(b_seq, tgt)
+    CZ(b_seq, ctrl, tgt)
+    Y_2(b_seq, ctrl)
+    X_2(b_seq, tgt)
+
+
+def SWAP(b_seq: Baking, ctrl: str = "q0", tgt: str = "q1"):
+    CNOT(b_seq, ctrl, tgt)
+    CNOT(b_seq, tgt, ctrl)
+    CNOT(b_seq, ctrl, tgt)
+
+
+"""
+In  what follows, q_tgt should be the main target qubit for which should be played the single qubit gate.
+If other elements shall be used to compute a single qubit gate, it is possible to modify/add methods related to the resolver
+to retrieve easily each element required.
+"""
+
+
+def I(b: Baking, q_tgt):
+    b.play("I", resolve.q(q_tgt))
+
+
+def X(b: Baking, q_tgt):
+    b.play("X", resolve.q(q_tgt))
+
+
+def Y(b: Baking, q_tgt):
+    b.play("Y", resolve.q(q_tgt))
+
+
+def X_2(b: Baking, q_tgt):
+    b.play("X/2", resolve.q(q_tgt))
+
+
+def Y_2(b: Baking, q_tgt):
+    b.play("Y/2", resolve.q(q_tgt))
+
+
+def mX_2(b: Baking, q_tgt):
+    b.play("-X/2", resolve.q(q_tgt))
+
+
+def mY_2(b: Baking, q_tgt):
+    b.play("-Y/2", resolve.q(q_tgt))
+
+
+two_qb_gate_macros = {
```

Comment:
how do I know if I implemented all the required gates? Perhaps this should be a dataclass or namedtuple which forces the user to provide all required arguments when instantiating?

Reply:
I could add further assertion in the class to make sure the correct gates are provided. Not sure to understand what you mean by dataclass or namedtuple in this case.